nucypher/blockchain/eth/actors.py

```diff
@@ -623,11 +623,10 @@ def perform_round_1(
                 ritual_id=ritual.id,
             )
         except Exception as e:
-            # TODO: Handle this better #3096
             self.log.critical(
                 f"Failed to generate a transcript for ritual #{ritual.id}: {str(e)}"
             )
-            raise e
+            return
```

Comment:
Note for reviewers - It may not be obvious what's happening here.  If this line raises, it ultimately crashes the async looping call that dispatches this task without removing it from the queue which results in a re-attempt on the next iteration. 

By returning here instead, we allow the task to complete without errors which removes this task from the queue.

Reply:
> Question: is it better to log a full stack trace here for log visibility (same below)?

Given that it is critical - we should log the stack trace. Same for below. wdyt?

Reply:
Side note: could be good to add a simple unit/integration test that ensures `perform_round_1`/`perform_round_2` return safely, without raising, for a cryptographic exception generated in the try-block.